src/common/checksum.c

```diff
@@ -707,26 +734,29 @@ calc_for_iov(struct daos_csummer *csummer, daos_key_t *iov,
 	memset(csum_buf, 0, csum_buf_len);
 
 	daos_csummer_set_buffer(csummer, csum_buf, csum_buf_len);
-
+	D_MUTEX_LOCK(&csummer->dcs_lock);
```

Comment:
why do we need locking for csummer?

Reply:
The csummer uses a single internal buffer that I believe multiple threads can update and cause a race condition.